hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -361,15 +360,13 @@ estimateScriptsCost ::
   Map TxIn TxOut ->
   -- | The pre-constructed transaction
   ValidatedTx LedgerEra ->
-  Either (RdmrPtr, TransactionScriptFailure StandardCrypto) (Map RdmrPtr ExUnits)
+  Either ErrCoverFee (Map RdmrPtr ExUnits)
 estimateScriptsCost pparams systemStart epochInfo utxo tx = do
-  -- FIXME: throwing exceptions in pure code is discouraged! Convert them to
-  -- throwM or throwIO or represent thes situations in the return type!
   case result of
     Left translationError ->
-      throw $ BadTranslationException translationError
+      Left $ ErrTranslationError translationError
```

Comment:
This is better. However, `ErrCoverFee` is a bit of a kitchen sink. The next step would be to have a dedicated error type for this function or finally bite the bullet to refactor the whole module to properly use cardano-api types.